packages/ui/src/working-groups/modals/ApplyForRoleModal/ApplyForRoleModal.tsx

```diff
@@ -108,41 +122,50 @@ export const ApplyForRoleModal = () => {
   const bindStakingAccountService = state.children.bindStakingAccount
 
   if (state.matches('bindStakingAccount') && api && bindStakingAccountService) {
-    const { stake } = state.context
-    const stakingAccount = stake.account?.address
-
-    const transaction = api.tx.members.addStakingAccountCandidate(active.id)
+    const stakingAccount = state.context.stake.account?.address
+    const transaction = api.tx.members.addStakingAccountCandidate(activeMember.id)
 
     return (
       <BindStakingAccountModal
         onClose={hideModal}
         transaction={transaction}
         signer={stakingAccount}
-        memberId={active.id}
+        memberId={activeMember.id}
         service={bindStakingAccountService}
         steps={transactionsSteps}
       />
     )
   }
 
-  const signer = active?.controllerAccount
+  const signer = activeMember?.controllerAccount
   const transactionService = state.children.transaction
 
   if (state.matches('transaction') && signer && api && transactionService) {
     const { stake, answers } = state.context
 
-    const transaction = getGroup(api, opening.groupName as GroupName)?.applyOnOpening({
+    const applyOnOpeningTransaction = getGroup(api, opening.groupName as GroupName)?.applyOnOpening({
       opening_id: opening.runtimeId,
-      member_id: active?.id,
-      role_account_id: active?.controllerAccount,
-      reward_account_id: active?.rootAccount,
+      member_id: activeMember?.id,
+      role_account_id: activeMember?.controllerAccount,
+      reward_account_id: activeMember?.rootAccount,
       description: metadataToBytes(ApplicationMetadata, { answers: Object.values(answers) }),
       stake_parameters: {
         stake: stake.amount,
         stake_account_id: stake.account?.address,
       },
     })
 
+    let transaction: SubmittableExtrinsic<'rxjs', ISubmittableResult>
+
+    if (stakingStatus === 'confirmed') {
+      transaction = applyOnOpeningTransaction
+    } else {
+      transaction = api.tx.utility.batch([
+        applyOnOpeningTransaction,
+        api.tx.members.confirmStakingAccount(activeMember?.id, stake.account.address),
```

Comment:
How does batch processing work exactly? I'm assuming it's sequential - shouldn't the `confirmStakingAccount` transaction be first in batch?

Reply:
Oh... nice catch :D

Reply:
I'd assume that it should work as you described.